tensorizer/stream_io.py

```diff
@@ -531,7 +562,41 @@ def seek(self, position, whence=SEEK_SET):
             self.close()
 
             # And we reinitialize ourself.
-            self.__init__(self._uri, position, None)
+            self.__init__(
+                uri=self._uri,
+                begin=position,
+                end=self._end,
+                headers=self._headers,
+                buffer_size=self._buffer_size,
+                certificate_handling=self._certificate_handling,
+            )
+
+    def _fork(
+        self, begin: Optional[int] = None, end: Optional[int] = None
+    ) -> "CURLStreamFile":
+        """
+        Reopen a file at a new byte range. Meant only for internal use
+        by `TensorDeserializer` when creating multiple readers.
+
+        Args:
+            begin: Starting byte of the new byte range, inclusive,
+                or None to start at the beginning of the file.
+            end: Ending byte of the new byte range, inclusive,
+                or None to include up to the end of the file.
+
+        Returns:
+            A new `CURLStreamFile` identical to this one,
+            but with a different byte range.
+        """
+        clone = self.__class__(
+            uri=self._uri,
+            begin=begin,
+            end=end,
+            headers=self._headers,
+            buffer_size=self._buffer_size,
+        )
```

Comment:
Are we not passing `certificate_handling`?

Reply:
What ever do you mean? Of course we're passing `certificate_handling` 😇 (fixed in b6f2697555c86fe998d8ff9ff18b8e764f8f4612)